lnbits/wallets/nwc.py

```diff
@@ -0,0 +1,881 @@
+        **kwargs
+    ) -> InvoiceResponse:
+        desc = ""
+        desc_hash = None
+        if description_hash:
+            desc_hash = description_hash.hex()
+            desc = (unhashed_description or b"").decode()
+        elif unhashed_description:
+            desc = unhashed_description.decode()
+            desc_hash = hashlib.sha256(desc.encode()).hexdigest()
+        else:
+            desc = memo or ""
+        try:
+            info = await self._get_info()
+            if "make_invoice" not in info["supported_methods"]:
+                return InvoiceResponse(
+                    False,
+                    None,
+                    None,
+                    "make_invoice is not supported by this NWC service.",
+                )
+            resp = await self._call(
+                "make_invoice",
+                {
+                    "amount": int(amount * 1000),  # nwc uses msats denominations
+                    "description_hash": desc_hash,
+                    "description": desc,
+                },
+            )
+            checking_id = str(resp["payment_hash"])
+            payment_request = resp.get("invoice", None)
+            # if lookup_invoice is not supported, we can't track the payment
+            if "lookup_invoice" in info["supported_methods"]:
+                created_at = int(resp.get("created_at", time.time()))
+                expires_at = int(resp.get("expires_at", created_at + 3600))
+                self.pending_payments.append(
+                    {  # Start tracking
+                        "checking_id": checking_id,
+                        "expires_at": expires_at,
+                        "settled": False,
+                    }
+                )
+            return InvoiceResponse(True, checking_id, payment_request, None)
+        except Exception as e:
+            return InvoiceResponse(ok=False, error_message=str(e))
+
+    async def status(self) -> StatusResponse:
+        try:
+            info = await self._get_info()
+            if "get_balance" not in info["supported_methods"]:
+                logger.debug("get_balance is not supported by this NWC service.")
+                return StatusResponse(None, 0)
+            resp = await self._call("get_balance", {})
+            balance = int(resp["balance"])
+            return StatusResponse(None, balance)
+        except Exception as e:
+            return StatusResponse(str(e), 0)
+
+    async def pay_invoice(self, bolt11: str, fee_limit_msat: int) -> PaymentResponse:
+        try:
+            resp = await self._call("pay_invoice", {"invoice": bolt11})
+            preimage = resp.get("preimage", None)
+            invoice_data = bolt11_decode(bolt11)
+            payment_hash = invoice_data.payment_hash
+            # pay_invoice doesn't return payment data, so we need
+            # to call lookup_invoice too (if supported)
+            info = await self._get_info()
```

Comment:
If we remove the workaround, then yes, the logic is pretty much the same.
But since the specs do not explicitly require the payment to be immediately available to be queried with lookup_invoice,  i suggest to leave this extra logic only on pay_invoice (would be wrong to add it to get_payment_status)

Reply:
> if we remove the workaround, then yes, the logic is pretty much the same.

By workaround do you mean the try-catch?

Reply:
yes